### PR TITLE
feat: create a dev container that contains all the dev tools needed to build nexodus.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

**Comment by russellb** [comment]
I was thinking of adding more docs in docs/development to cover a development workflow overview. Just sharing in case you feel inspired to write a blurb on how to use this.

File: Makefile [hunk 1]
```diff
@@ -227,6 +227,25 @@ run-test-container: ## Run docker container that you can run nexodus in
 		--mount type=bind,source=$(shell pwd)/.certs,target=/.certs,readonly \
 		quay.io/nexodus/test:$(TEST_CONTAINER_DISTRO) /update-ca.sh
 
+.dev-container: Containerfile.dev
+	$(CMD_PREFIX) docker build -f Containerfile.dev -t quay.io/nexodus/dev:latest .
+	$(CMD_PREFIX) touch $@
+
+.PHONY: run-dev-container
+run-dev-container: .dev-container ## Run docker container that you can develop and run nexodus in
```
Comment (russellb) [inline Makefile:235]: Do we still need run-test-container or can this solve both use cases?

Comment (chirino) [reply Makefile:235]: No.. so this container has developer tooling.  I think test container should get replaced by the nexd container which is basically just an image with the nexd/nexctl runtimes.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "nexodus build tools",
+  "build": {
+    "dockerfile": "../Containerfile.dev"
+  }
```
[the remark this answers is not in the document]
Comment (chirino) [reply .devcontainer/devcontainer.json:5]: sadly things like kind don't work well yet.  kubectl get's configured with the wrong host name.

[the remark this answers is not in the document]
Comment (russellb) [reply .devcontainer/devcontainer.json:5]: Cool, I use vscode but hadn't used that feature. Is this optional? I tend to do everything in my vscode terminals, including dealing with kind. If we know kind is busted in this env, can we detect it in the Makefile?

File: Makefile (hunk 1 above)
Comment (nerdalert) [reply Makefile:235]: Rational behind having a bin/sh container with nexd copied in is for testing killing and respawning nexd.

File: .devcontainer/devcontainer.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (chirino) [reply .devcontainer/devcontainer.json:5]: Yeah we can likely do something to know we are in there

**Comment by russellb** [comment]
I played with this locally a bit and verified that it's optional with vscode, so I'm good with it.

**Pull request merged**